identity/middleware.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package identity
+
+import (
+	"net/http"
+
+	"github.com/ant0ine/go-json-rest/rest"
+
+	"github.com/mendersoftware/go-lib-micro/log"
+	"github.com/mendersoftware/go-lib-micro/rest_utils"
+	"github.com/pkg/errors"
+)
+
+// IdentityMiddleware adds the identity extracted from JWT token to the request's context.
+type IdentityMiddleware struct {
```

Comment:
And another note that the middleware should be added after requestlog and requestid middlewares.

Also, it should be after context repacking middleware (since we're using a request logger from context).